# COVID-19 Tests

Jimuel Celeste, Jr. 

Objective: To estimate the availability of tests in provinces through the number of tests per province.

In [87]:
import pandas as pd

## COVID-19 Testing Facilities

In [88]:
facilities = "/Users/jimuelcelestejr/Documents/codebook/covid-asmr-philippines/data/COVID-19 Tests/COVID testing facilities (1).xlsx"
facilities = pd.read_excel(facilities)

In [89]:
facilities

ID  REGION             PROVINCE  \
0      1.0     NCR         Metro Manila   
1      2.0     NCR         Metro Manila   
2      3.0  CARAGA      Surigao del Sur   
3      4.0     NCR         Metro Manila   
4      5.0     NCR         Metro Manila   
..     ...     ...                  ...   
391  348.0      R9    Zamboanga del Sur   
392    NaN      R9    Zamboanga del Sur   
393    NaN      R9    Zamboanga del Sur   
394  349.0      R9  Zamboanga del Norte   
395  350.0      R9    Zamboanga del Sur   

                                  EDITED FACILITY NAME  \
0                                  a star laboratories   
1                                                  NaN   
2                                                  NaN   
3                                                  NaN   
4       al molecular diagnostic laboratory - genexpert   
..                                                 ...   
391  zamboanga city medical center genexpert labora...   
392       zamboanga city medical center  (plate-based)   
393  zamboanga city medical center - da satellite l...   
394                                                NaN   
395    zamboanga del sur medical center pcr laboratory   

                                              FACILITY        TYPE  \
0                             A Star Laboratories, Inc     Private   
1                   AC Molecular Diagnostic Laboratory     Private   
2               Adela Serra TY Memorial Medical Center  Government   
3                   Al Molecular Diagnostic Laboratory     Private   
4                   Al Molecular Diagnostic Laboratory     Private   
..                                                 ...         ...   
391  Zamboanga City Medical Center - NTP Culture La...  Government   
392                                                NaN         NaN   
393                                                NaN         NaN   
394                 Zamboanga del Norte Medical Center  Government   
395                   Zamboanga del Sur Medical Center  Government   

                      TEST Remarks  
0                  rRT PCR     NaN  
1                  rRT PCR     NaN  
2                  rRT PCR     NaN  
3                  rRT PCR     NaN  
4    Cartridge - Based PCR     NaN  
..                     ...     ...  
391  Cartridge - Based PCR     NaN  
392                    NaN     NaN  
393                    NaN     NaN  
394                rRT PCR     NaN  
395                rRT PCR     NaN  

[396 rows x 8 columns]

In [90]:
tests = "/Users/jimuelcelestejr/Documents/codebook/covid-asmr-philippines/data/COVID-19 Tests/DOH COVID Data Drop_ 20240103 - 07 Testing Aggregates.csv"
tests = pd.read_csv(tests)

/var/folders/ny/jv8cwgpx7ng41dfgnqsvzdnw0000gn/T/ipykernel_23663/2228410702.py:2: DtypeWarning: Columns (17) have mixed types. Specify dtype option on import or set low_memory=False.
  tests = pd.read_csv(tests)


In [91]:
tests

facility_name report_date  avg_turnaround_time  \
0              A Star Laboratories  2020-12-08                 24.0   
1              A Star Laboratories  2020-12-09                  NaN   
2              A Star Laboratories  2020-12-10                  NaN   
3              A Star Laboratories  2020-12-11                  NaN   
4              A Star Laboratories  2020-12-12                  NaN   
...                            ...         ...                  ...   
393640  truhealthencoder@gmail.com  2023-12-28                  NaN   
393641  truhealthencoder@gmail.com  2023-12-29                  NaN   
393642  truhealthencoder@gmail.com  2023-12-30                  NaN   
393643  truhealthencoder@gmail.com  2023-12-31                  NaN   
393644  truhealthencoder@gmail.com  2024-01-01                  NaN   

        daily_output_samples_tested  daily_output_unique_individuals  \
0                              15.0                             15.0   
1                               NaN                              NaN   
2                               NaN                              NaN   
3                               NaN                              NaN   
4                               NaN                              NaN   
...                             ...                              ...   
393640                          NaN                              NaN   
393641                          NaN                              NaN   
393642                          NaN                              NaN   
393643                          NaN                              NaN   
393644                          NaN                              NaN   

        daily_output_positive_individuals  daily_output_negative_individuals  \
0                                     1.0                               14.0   
1                                     NaN                                NaN   
2                                     NaN                                NaN   
3                                     NaN                                NaN   
4                                     NaN                                NaN   
...                                   ...                                ...   
393640                                NaN                                NaN   
393641                                NaN                                NaN   
393642                                NaN                                NaN   
393643                                NaN                                NaN   
393644                                NaN                                NaN   

        daily_output_equivocal  daily_output_invalid  \
0                          0.0                   0.0   
1                          NaN                   NaN   
2                          NaN                   NaN   
3                          NaN                   NaN   
4                          NaN                   NaN   
...                        ...                   ...   
393640                     NaN                   NaN   
393641                     NaN                   NaN   
393642                     NaN                   NaN   
393643                     NaN                   NaN   
393644                     NaN                   NaN   

        remaining_available_tests  backlogs  cumulative_samples_tested  \
0                             8.0       0.0                         15   
1                             NaN       NaN                         15   
2                             NaN       NaN                         15   
3                             NaN       NaN                         15   
4                             NaN       NaN                         15   
...                           ...       ...                        ...   
393640                        NaN       NaN                         12   
393641                        NaN       NaN                         12   
393

In [92]:
first_list = [facility.lower() for facility in tests['facility_name'].unique()]
first_list

['a star laboratories',
 'ac molecular diagnostic laboratory',
 'al molecular diagnostic laboratory',
 'al molecular diagnostic laboratory - genexpert',
 'apex mining co. inc.',
 'adela serra ty memorial medical center',
 'allegiant regional care hospital',
 'allegiant regional care hospital - cartridge based',
 'allied care expert - valenzuela',
 'allied medical and molecular laboratory corp.',
 'amai pakpak medical center genexpert laboratory',
 'amai pakpak medical center molecular laboratory',
 'amang rodriguez memoral medical center (rt pcr)',
 'amang rodriguez memorial center genexpert laboratory',
 "amosup seamen's hospital",
 'antipolo city molecular laboratory',
 'antique bio-molecular laboratory',
 'army general hospital molecular laboratory',
 'asian hospital and medical center',
 'asian hospital and medical center - cartridge',
 'asian hospital and medical center - rt-pcr',
 'ateneo molecular pathology laboratory',
 'brtth covid-19 testing laboratoy',
 'bacolod queen of mer

In [94]:
second_list = [facility.lower() for facility in facilities['FACILITY'].dropna().unique()]
second_list

['a star laboratories, inc',
 'ac molecular diagnostic laboratory',
 'adela serra ty memorial medical center',
 'al molecular diagnostic laboratory',
 'allegiant regional care hospital',
 'allied care expert (ace) medical center valenzuela, inc',
 'allied medical and molecular laboratory corp.',
 'amai pakpak medical center',
 'amang rodriguez memorial center',
 'amang rodriguez memorial hospital',
 'antipolo city molecular laboratory',
 'antique bio-molecular laboratory',
 'apex mining co. inc.',
 'army general hospita',
 'asian hospital inc.',
 'bacolod queen of mercy hospital',
 'baguio general hospital and medical center',
 'bataan general hospital',
 'batangas medical center',
 'be safe md, inc.',
 'benguet general hospital',
 'best diagnostic corporation',
 'bicol-south luzon sub national reference laboratory',
 'bicol medical center',
 'bicol region general hospital and geriatric medical center',
 'biopath clinical diagnostic, inc.',
 'bohol containerized pcr laboratory',
 'bree

Symmetric Difference
See: https://www.scaler.com/topics/python-difference-between-two-lists/

In [95]:
difference_result = set(first_list).symmetric_difference(set(second_list))
list_difference_result = list(difference_result)

print(list_difference_result)

['dr. fe del mundo medical center', 'm diagnostics philippines, inc. - pasay (cartridge)', 'next gen pcr diagnostic laboratory - cebu', 'mariano marcos memorial hospital (rt pcr)', 'maguindanao provincial hospital - rt-pcr', 'the medical city clark', 'makati medical center (hb)', 'northern mindanao medical center', 'bataan general hospital', 'greencity medical center', 'cagayan valley medical center, inc.', 'health metrics', 'research institute for tropical medicine, inc. (ritm)', 'university of perpetual help dalta medical center, inc.', 'cagayan valley medical center molecular laboratory - cartridge-based', 'laguna holy family hospital, inc. - satellite molecular laboratory', 'ospital ng parañaque district ii', 'metro north medical center and hospital inc.', 'jose b. lingad memorial hospital', 'um medical & multitest diagnostic inc., molecular laboratory', 'dr. jose fabella memorial hospital', 'cabanatuan city m. v. gallego general hospital', 'northern mindanao tb regional center', '

In [96]:
len(list_difference_result)

331

331 is a lot to manually annotate. It might be easier to edit only one file, i.e., the file containing information about the testing facilities.

In [97]:
len(first_list)

387

In [98]:
len(second_list)

286

In [99]:
# Facilities that are listed in the COVID-19 tests performed dataset
# that are not in the COVID-19 testing center dataset.
i = 0
for facility in first_list:
    if facility not in second_list:
        print(i, facility)
        i += 1

0 a star laboratories
1 al molecular diagnostic laboratory - genexpert
2 allegiant regional care hospital - cartridge based
3 allied care expert - valenzuela
4 amai pakpak medical center genexpert laboratory
5 amai pakpak medical center molecular laboratory
6 amang rodriguez memoral medical center (rt pcr)
7 amang rodriguez memorial center genexpert laboratory
8 amosup seamen's hospital
9 army general hospital molecular laboratory
10 asian hospital and medical center
11 asian hospital and medical center - cartridge
12 asian hospital and medical center - rt-pcr
13 ateneo molecular pathology laboratory
14 brtth covid-19 testing laboratoy
15 bataan general hospital (rt pcr)
16 bataan general hospital and medical center genexpert laboratory
17 batangas medical center genexpert laboratory
18 batangas medical center rt pcr laboratory
19 be safe md, inc. - cartridge based
20 bicol medical center genexpert laboratory
21 bicol medical center rt-pcr laboratory
22 bicol regional diagnostic and re

There are 215 such records.

Objective: Manually edit the file to consolidate the records. Add a new column named "EDITED FACILITY NAME"

Rules:
1. Change spelling if there are differences in spelling. 
2. Remove additional words if needed, i.e., "A Star Laboratories, Inc" to "A Start Laboratories" 
3. Add facilities to the list if not included, i.e., Ateneo Molecular Pathology Laboratory

Note:
1. One health facility is not added to the list of facilities, i.e., truhealthencoder@gmail.com. We do not know which province to include this testing facility.

In [100]:
# Edited version of the file (same file as above with additional column)
facilities_2 = "/Users/jimuelcelestejr/Documents/codebook/covid-asmr-philippines/data/COVID-19 Tests/COVID testing facilities (1).xlsx"
facilities_2 = pd.read_excel(facilities_2)

In [101]:
facilities_2

ID  REGION             PROVINCE  \
0      1.0     NCR         Metro Manila   
1      2.0     NCR         Metro Manila   
2      3.0  CARAGA      Surigao del Sur   
3      4.0     NCR         Metro Manila   
4      5.0     NCR         Metro Manila   
..     ...     ...                  ...   
391  348.0      R9    Zamboanga del Sur   
392    NaN      R9    Zamboanga del Sur   
393    NaN      R9    Zamboanga del Sur   
394  349.0      R9  Zamboanga del Norte   
395  350.0      R9    Zamboanga del Sur   

                                  EDITED FACILITY NAME  \
0                                  a star laboratories   
1                                                  NaN   
2                                                  NaN   
3                                                  NaN   
4       al molecular diagnostic laboratory - genexpert   
..                                                 ...   
391  zamboanga city medical center genexpert labora...   
392       zamboanga city medical center  (plate-based)   
393  zamboanga city medical center - da satellite l...   
394                                                NaN   
395    zamboanga del sur medical center pcr laboratory   

                                              FACILITY        TYPE  \
0                             A Star Laboratories, Inc     Private   
1                   AC Molecular Diagnostic Laboratory     Private   
2               Adela Serra TY Memorial Medical Center  Government   
3                   Al Molecular Diagnostic Laboratory     Private   
4                   Al Molecular Diagnostic Laboratory     Private   
..                                                 ...         ...   
391  Zamboanga City Medical Center - NTP Culture La...  Government   
392                                                NaN         NaN   
393                                                NaN         NaN   
394                 Zamboanga del Norte Medical Center  Government   
395                   Zamboanga del Sur Medical Center  Government   

                      TEST Remarks  
0                  rRT PCR     NaN  
1                  rRT PCR     NaN  
2                  rRT PCR     NaN  
3                  rRT PCR     NaN  
4    Cartridge - Based PCR     NaN  
..                     ...     ...  
391  Cartridge - Based PCR     NaN  
392                    NaN     NaN  
393                    NaN     NaN  
394                rRT PCR     NaN  
395                rRT PCR     NaN  

[396 rows x 8 columns]

In [102]:
facilities_2['Facility_Names_Merged'] = facilities_2['EDITED FACILITY NAME']
facilities_2

ID  REGION             PROVINCE  \
0      1.0     NCR         Metro Manila   
1      2.0     NCR         Metro Manila   
2      3.0  CARAGA      Surigao del Sur   
3      4.0     NCR         Metro Manila   
4      5.0     NCR         Metro Manila   
..     ...     ...                  ...   
391  348.0      R9    Zamboanga del Sur   
392    NaN      R9    Zamboanga del Sur   
393    NaN      R9    Zamboanga del Sur   
394  349.0      R9  Zamboanga del Norte   
395  350.0      R9    Zamboanga del Sur   

                                  EDITED FACILITY NAME  \
0                                  a star laboratories   
1                                                  NaN   
2                                                  NaN   
3                                                  NaN   
4       al molecular diagnostic laboratory - genexpert   
..                                                 ...   
391  zamboanga city medical center genexpert labora...   
392       zamboanga city medical center  (plate-based)   
393  zamboanga city medical center - da satellite l...   
394                                                NaN   
395    zamboanga del sur medical center pcr laboratory   

                                              FACILITY        TYPE  \
0                             A Star Laboratories, Inc     Private   
1                   AC Molecular Diagnostic Laboratory     Private   
2               Adela Serra TY Memorial Medical Center  Government   
3                   Al Molecular Diagnostic Laboratory     Private   
4                   Al Molecular Diagnostic Laboratory     Private   
..                                                 ...         ...   
391  Zamboanga City Medical Center - NTP Culture La...  Government   
392                                                NaN         NaN   
393                                                NaN         NaN   
394                 Zamboanga del Norte Medical Center  Government   
395                   Zamboanga del Sur Medical Center  Government   

                      TEST Remarks  \
0                  rRT PCR     NaN   
1                  rRT PCR     NaN   
2                  rRT PCR     NaN   
3                  rRT PCR     NaN   
4    Cartridge - Based PCR     NaN   
..                     ...     ...   
391  Cartridge - Based PCR     NaN   
392                    NaN     NaN   
393                    NaN     NaN   
394                rRT PCR     NaN   
395                rRT PCR     NaN   

                                 Facility_Names_Merged  
0                                  a star laboratories  
1                                                  NaN  
2                                                  NaN  
3                                                  NaN  
4       al molecular diagnostic laboratory - genexpert  
..                                                 ...  
391  zamboanga city medical center genexpert labora...  
392       zamboanga city medical center  (plate-based)  
393  zamboanga city medical center - da satellite l...  
394                                                NaN  
395    zamboanga del sur medical center pcr laboratory  

[396 rows x 9 columns]

In [103]:
index_empty = facilities_2['Facility_Names_Merged'].isna()
facilities_2.loc[index_empty, 'Facility_Names_Merged'] = facilities_2[index_empty]['FACILITY']

In [104]:
facilities_2

ID  REGION             PROVINCE  \
0      1.0     NCR         Metro Manila   
1      2.0     NCR         Metro Manila   
2      3.0  CARAGA      Surigao del Sur   
3      4.0     NCR         Metro Manila   
4      5.0     NCR         Metro Manila   
..     ...     ...                  ...   
391  348.0      R9    Zamboanga del Sur   
392    NaN      R9    Zamboanga del Sur   
393    NaN      R9    Zamboanga del Sur   
394  349.0      R9  Zamboanga del Norte   
395  350.0      R9    Zamboanga del Sur   

                                  EDITED FACILITY NAME  \
0                                  a star laboratories   
1                                                  NaN   
2                                                  NaN   
3                                                  NaN   
4       al molecular diagnostic laboratory - genexpert   
..                                                 ...   
391  zamboanga city medical center genexpert labora...   
392       zamboanga city medical center  (plate-based)   
393  zamboanga city medical center - da satellite l...   
394                                                NaN   
395    zamboanga del sur medical center pcr laboratory   

                                              FACILITY        TYPE  \
0                             A Star Laboratories, Inc     Private   
1                   AC Molecular Diagnostic Laboratory     Private   
2               Adela Serra TY Memorial Medical Center  Government   
3                   Al Molecular Diagnostic Laboratory     Private   
4                   Al Molecular Diagnostic Laboratory     Private   
..                                                 ...         ...   
391  Zamboanga City Medical Center - NTP Culture La...  Government   
392                                                NaN         NaN   
393                                                NaN         NaN   
394                 Zamboanga del Norte Medical Center  Government   
395                   Zamboanga del Sur Medical Center  Government   

                      TEST Remarks  \
0                  rRT PCR     NaN   
1                  rRT PCR     NaN   
2                  rRT PCR     NaN   
3                  rRT PCR     NaN   
4    Cartridge - Based PCR     NaN   
..                     ...     ...   
391  Cartridge - Based PCR     NaN   
392                    NaN     NaN   
393                    NaN     NaN   
394                rRT PCR     NaN   
395                rRT PCR     NaN   

                                 Facility_Names_Merged  
0                                  a star laboratories  
1                   AC Molecular Diagnostic Laboratory  
2               Adela Serra TY Memorial Medical Center  
3                   Al Molecular Diagnostic Laboratory  
4       al molecular diagnostic laboratory - genexpert  
..                                                 ...  
391  zamboanga city medical center genexpert labora...  
392       zamboanga city medical center  (plate-based)  
393  zamboanga city medical center - da satellite l...  
394                 Zamboanga del Norte Medical Center  
395    zamboanga del sur medical center pcr laboratory  

[396 rows x 9 columns]

In [105]:
output_file = "/Users/jimuelcelestejr/Documents/codebook/covid-asmr-philippines/data/COVID-19 Tests/facilities_2.csv"
facilities_2.to_csv(output_file, index=False)

In [106]:
facility_df = facilities_2[['Facility_Names_Merged', 'PROVINCE']].rename(columns={
    'Facility_Names_Merged': 'Facility', 
    'PROVINCE': 'Province'
})

In [107]:
facility_df['Facility'] = facility_df['Facility'].apply(lambda x: x.lower())
facility_df['Province'] = facility_df['Province'].apply(lambda x: x.upper())

In [108]:
facility_df

Facility             Province
0                                  a star laboratories         METRO MANILA
1                   ac molecular diagnostic laboratory         METRO MANILA
2               adela serra ty memorial medical center      SURIGAO DEL SUR
3                   al molecular diagnostic laboratory         METRO MANILA
4       al molecular diagnostic laboratory - genexpert         METRO MANILA
..                                                 ...                  ...
391  zamboanga city medical center genexpert labora...    ZAMBOANGA DEL SUR
392       zamboanga city medical center  (plate-based)    ZAMBOANGA DEL SUR
393  zamboanga city medical center - da satellite l...    ZAMBOANGA DEL SUR
394                 zamboanga del norte medical center  ZAMBOANGA DEL NORTE
395    zamboanga del sur medical center pcr laboratory    ZAMBOANGA DEL SUR

[396 rows x 2 columns]

In [109]:
output_file = "/Users/jimuelcelestejr/Documents/codebook/covid-asmr-philippines/data/COVID-19 Tests/testing_facilities.csv"
facility_df.to_csv(output_file, index=False)

## COVID-19 Tests Performed per Facility

In [110]:
tests

facility_name report_date  avg_turnaround_time  \
0              A Star Laboratories  2020-12-08                 24.0   
1              A Star Laboratories  2020-12-09                  NaN   
2              A Star Laboratories  2020-12-10                  NaN   
3              A Star Laboratories  2020-12-11                  NaN   
4              A Star Laboratories  2020-12-12                  NaN   
...                            ...         ...                  ...   
393640  truhealthencoder@gmail.com  2023-12-28                  NaN   
393641  truhealthencoder@gmail.com  2023-12-29                  NaN   
393642  truhealthencoder@gmail.com  2023-12-30                  NaN   
393643  truhealthencoder@gmail.com  2023-12-31                  NaN   
393644  truhealthencoder@gmail.com  2024-01-01                  NaN   

        daily_output_samples_tested  daily_output_unique_individuals  \
0                              15.0                             15.0   
1                               NaN                              NaN   
2                               NaN                              NaN   
3                               NaN                              NaN   
4                               NaN                              NaN   
...                             ...                              ...   
393640                          NaN                              NaN   
393641                          NaN                              NaN   
393642                          NaN                              NaN   
393643                          NaN                              NaN   
393644                          NaN                              NaN   

        daily_output_positive_individuals  daily_output_negative_individuals  \
0                                     1.0                               14.0   
1                                     NaN                                NaN   
2                                     NaN                                NaN   
3                                     NaN                                NaN   
4                                     NaN                                NaN   
...                                   ...                                ...   
393640                                NaN                                NaN   
393641                                NaN                                NaN   
393642                                NaN                                NaN   
393643                                NaN                                NaN   
393644                                NaN                                NaN   

        daily_output_equivocal  daily_output_invalid  \
0                          0.0                   0.0   
1                          NaN                   NaN   
2                          NaN                   NaN   
3                          NaN                   NaN   
4                          NaN                   NaN   
...                        ...                   ...   
393640                     NaN                   NaN   
393641                     NaN                   NaN   
393642                     NaN                   NaN   
393643                     NaN                   NaN   
393644                     NaN                   NaN   

        remaining_available_tests  backlogs  cumulative_samples_tested  \
0                             8.0       0.0                         15   
1                             NaN       NaN                         15   
2                             NaN       NaN                         15   
3                             NaN       NaN                         15   
4                             NaN       NaN                         15   
...                           ...       ...                        ...   
393640                        NaN       NaN                         12   
393641                        NaN       NaN                         12   
393

In [111]:
tests[tests['facility_name']=='A Star Laboratories']

facility_name report_date  avg_turnaround_time  \
0     A Star Laboratories  2020-12-08                 24.0   
1     A Star Laboratories  2020-12-09                  NaN   
2     A Star Laboratories  2020-12-10                  NaN   
3     A Star Laboratories  2020-12-11                  NaN   
4     A Star Laboratories  2020-12-12                  NaN   
...                   ...         ...                  ...   
1115  A Star Laboratories  2023-12-28                 12.0   
1116  A Star Laboratories  2023-12-29                 12.0   
1117  A Star Laboratories  2023-12-30                 12.0   
1118  A Star Laboratories  2023-12-31                 12.0   
1119  A Star Laboratories  2024-01-01                  NaN   

      daily_output_samples_tested  daily_output_unique_individuals  \
0                            15.0                             15.0   
1                             NaN                              NaN   
2                             NaN                              NaN   
3                             NaN                              NaN   
4                             NaN                              NaN   
...                           ...                              ...   
1115                         11.0                             11.0   
1116                          7.0                              7.0   
1117                          5.0                              5.0   
1118                          0.0                              0.0   
1119                          NaN                              NaN   

      daily_output_positive_individuals  daily_output_negative_individuals  \
0                                   1.0                               14.0   
1                                   NaN                                NaN   
2                                   NaN                                NaN   
3                                   NaN                                NaN   
4                                   NaN                                NaN   
...                                 ...                                ...   
1115                                1.0                               10.0   
1116                                2.0                                5.0   
1117                                2.0                                3.0   
1118                                0.0                                0.0   
1119                                NaN                                NaN   

      daily_output_equivocal  daily_output_invalid  remaining_available_tests  \
0                        0.0                   0.0                        8.0   
1                        NaN                   NaN                        NaN   
2                        NaN                   NaN                        NaN   
3                        NaN                   NaN                        NaN   
4                        NaN                   NaN                        NaN   
...                      ...                   ...                        ...   
1115                     0.0                   0.0                      639.0   
1116                     0.0                   0.0                      630.0   
1117                     0.0                   0.0                      625.0   
1118                     0.0                   0.0                      625.0   
1119                     NaN                   NaN                        NaN   

      backlogs  cumulative_samples_tested  cumulative_unique_individuals  \
0          0.0                         15                             15   
1          NaN                         15                             15   
2          NaN                         15                             15   
3          NaN                         15                             15   
4          NaN                         15                             15   
...        ...                        ...                   

Filtering out records with report_date > 2023-05-09. The COVID-19 mortality data for this study were collected only until this date. 

In [112]:
tests['report_date'] = pd.to_datetime(tests['report_date'])
tests

facility_name report_date  avg_turnaround_time  \
0              A Star Laboratories  2020-12-08                 24.0   
1              A Star Laboratories  2020-12-09                  NaN   
2              A Star Laboratories  2020-12-10                  NaN   
3              A Star Laboratories  2020-12-11                  NaN   
4              A Star Laboratories  2020-12-12                  NaN   
...                            ...         ...                  ...   
393640  truhealthencoder@gmail.com  2023-12-28                  NaN   
393641  truhealthencoder@gmail.com  2023-12-29                  NaN   
393642  truhealthencoder@gmail.com  2023-12-30                  NaN   
393643  truhealthencoder@gmail.com  2023-12-31                  NaN   
393644  truhealthencoder@gmail.com  2024-01-01                  NaN   

        daily_output_samples_tested  daily_output_unique_individuals  \
0                              15.0                             15.0   
1                               NaN                              NaN   
2                               NaN                              NaN   
3                               NaN                              NaN   
4                               NaN                              NaN   
...                             ...                              ...   
393640                          NaN                              NaN   
393641                          NaN                              NaN   
393642                          NaN                              NaN   
393643                          NaN                              NaN   
393644                          NaN                              NaN   

        daily_output_positive_individuals  daily_output_negative_individuals  \
0                                     1.0                               14.0   
1                                     NaN                                NaN   
2                                     NaN                                NaN   
3                                     NaN                                NaN   
4                                     NaN                                NaN   
...                                   ...                                ...   
393640                                NaN                                NaN   
393641                                NaN                                NaN   
393642                                NaN                                NaN   
393643                                NaN                                NaN   
393644                                NaN                                NaN   

        daily_output_equivocal  daily_output_invalid  \
0                          0.0                   0.0   
1                          NaN                   NaN   
2                          NaN                   NaN   
3                          NaN                   NaN   
4                          NaN                   NaN   
...                        ...                   ...   
393640                     NaN                   NaN   
393641                     NaN                   NaN   
393642                     NaN                   NaN   
393643                     NaN                   NaN   
393644                     NaN                   NaN   

        remaining_available_tests  backlogs  cumulative_samples_tested  \
0                             8.0       0.0                         15   
1                             NaN       NaN                         15   
2                             NaN       NaN                         15   
3                             NaN       NaN                         15   
4                             NaN       NaN                         15   
...                           ...       ...                        ...   
393640                        NaN       NaN                         12   
393641                        NaN       NaN                         12   
393

In [113]:
tests = tests[tests['report_date'] <= pd.to_datetime(' 2023-05-09')]
tests

facility_name report_date  avg_turnaround_time  \
0              A Star Laboratories  2020-12-08                 24.0   
1              A Star Laboratories  2020-12-09                  NaN   
2              A Star Laboratories  2020-12-10                  NaN   
3              A Star Laboratories  2020-12-11                  NaN   
4              A Star Laboratories  2020-12-12                  NaN   
...                            ...         ...                  ...   
393403  truhealthencoder@gmail.com  2023-05-05                  NaN   
393404  truhealthencoder@gmail.com  2023-05-06                  NaN   
393405  truhealthencoder@gmail.com  2023-05-07                  NaN   
393406  truhealthencoder@gmail.com  2023-05-08                  NaN   
393407  truhealthencoder@gmail.com  2023-05-09                  NaN   

        daily_output_samples_tested  daily_output_unique_individuals  \
0                              15.0                             15.0   
1                               NaN                              NaN   
2                               NaN                              NaN   
3                               NaN                              NaN   
4                               NaN                              NaN   
...                             ...                              ...   
393403                          NaN                              NaN   
393404                          NaN                              NaN   
393405                          NaN                              NaN   
393406                          NaN                              NaN   
393407                          NaN                              NaN   

        daily_output_positive_individuals  daily_output_negative_individuals  \
0                                     1.0                               14.0   
1                                     NaN                                NaN   
2                                     NaN                                NaN   
3                                     NaN                                NaN   
4                                     NaN                                NaN   
...                                   ...                                ...   
393403                                NaN                                NaN   
393404                                NaN                                NaN   
393405                                NaN                                NaN   
393406                                NaN                                NaN   
393407                                NaN                                NaN   

        daily_output_equivocal  daily_output_invalid  \
0                          0.0                   0.0   
1                          NaN                   NaN   
2                          NaN                   NaN   
3                          NaN                   NaN   
4                          NaN                   NaN   
...                        ...                   ...   
393403                     NaN                   NaN   
393404                     NaN                   NaN   
393405                     NaN                   NaN   
393406                     NaN                   NaN   
393407                     NaN                   NaN   

        remaining_available_tests  backlogs  cumulative_samples_tested  \
0                             8.0       0.0                         15   
1                             NaN       NaN                         15   
2                             NaN       NaN                         15   
3                             NaN       NaN                         15   
4                             NaN       NaN                         15   
...                           ...       ...                        ...   
393403                        NaN       NaN                         12   
393404                        NaN       NaN                         12   
393

Taking the max "cumulative_samples_tested" per facility (group by facility_name).

https://scales.arabpsychology.com/stats/how-to-find-the-max-value-by-group-in-pandas/

In [114]:
tests_df = tests.groupby('facility_name')['cumulative_samples_tested'].max().reset_index()
tests_df

facility_name  \
0                                  A Star Laboratories   
1                   AC Molecular Diagnostic Laboratory   
2                   AL Molecular Diagnostic Laboratory   
3       AL Molecular Diagnostic Laboratory - GeneXpert   
4                                 APEX Mining CO. Inc.   
..                                                 ...   
380  Zamboanga City Medical Center - DA Satellite L...   
381  Zamboanga City Medical Center GeneXpert Labora...   
382                 Zamboanga del Norte Medical Center   
383    Zamboanga del Sur Medical Center PCR Laboratory   
384                         truhealthencoder@gmail.com   

     cumulative_samples_tested  
0                        86372  
1                        27634  
2                       235188  
3                         7292  
4                          262  
..                         ...  
380                      98686  
381                      56578  
382                      15342  
383                      44188  
384                         12  

[385 rows x 2 columns]

In [115]:
tests_df = tests_df.rename(columns={
    'facility_name': 'Facility', 
    'cumulative_samples_tested': 'Cumulative Samples Tested'
})
tests_df

Facility  \
0                                  A Star Laboratories   
1                   AC Molecular Diagnostic Laboratory   
2                   AL Molecular Diagnostic Laboratory   
3       AL Molecular Diagnostic Laboratory - GeneXpert   
4                                 APEX Mining CO. Inc.   
..                                                 ...   
380  Zamboanga City Medical Center - DA Satellite L...   
381  Zamboanga City Medical Center GeneXpert Labora...   
382                 Zamboanga del Norte Medical Center   
383    Zamboanga del Sur Medical Center PCR Laboratory   
384                         truhealthencoder@gmail.com   

     Cumulative Samples Tested  
0                        86372  
1                        27634  
2                       235188  
3                         7292  
4                          262  
..                         ...  
380                      98686  
381                      56578  
382                      15342  
383                      44188  
384                         12  

[385 rows x 2 columns]

In [116]:
tests_df['Facility'] = tests_df['Facility'].apply(lambda x: x.lower())
tests_df

Facility  \
0                                  a star laboratories   
1                   ac molecular diagnostic laboratory   
2                   al molecular diagnostic laboratory   
3       al molecular diagnostic laboratory - genexpert   
4                                 apex mining co. inc.   
..                                                 ...   
380  zamboanga city medical center - da satellite l...   
381  zamboanga city medical center genexpert labora...   
382                 zamboanga del norte medical center   
383    zamboanga del sur medical center pcr laboratory   
384                         truhealthencoder@gmail.com   

     Cumulative Samples Tested  
0                        86372  
1                        27634  
2                       235188  
3                         7292  
4                          262  
..                         ...  
380                      98686  
381                      56578  
382                      15342  
383                      44188  
384                         12  

[385 rows x 2 columns]

In [117]:
output_file = "/Users/jimuelcelestejr/Documents/codebook/covid-asmr-philippines/data/COVID-19 Tests/covid19_tests_performed.csv"
tests_df.to_csv(output_file, index=False)

## Merge 

1. Merge facility_df and test_df
2. Sum up the tests performed by province

In [118]:
facility_df

Facility             Province
0                                  a star laboratories         METRO MANILA
1                   ac molecular diagnostic laboratory         METRO MANILA
2               adela serra ty memorial medical center      SURIGAO DEL SUR
3                   al molecular diagnostic laboratory         METRO MANILA
4       al molecular diagnostic laboratory - genexpert         METRO MANILA
..                                                 ...                  ...
391  zamboanga city medical center genexpert labora...    ZAMBOANGA DEL SUR
392       zamboanga city medical center  (plate-based)    ZAMBOANGA DEL SUR
393  zamboanga city medical center - da satellite l...    ZAMBOANGA DEL SUR
394                 zamboanga del norte medical center  ZAMBOANGA DEL NORTE
395    zamboanga del sur medical center pcr laboratory    ZAMBOANGA DEL SUR

[396 rows x 2 columns]

In [119]:
tests_df

Facility  \
0                                  a star laboratories   
1                   ac molecular diagnostic laboratory   
2                   al molecular diagnostic laboratory   
3       al molecular diagnostic laboratory - genexpert   
4                                 apex mining co. inc.   
..                                                 ...   
380  zamboanga city medical center - da satellite l...   
381  zamboanga city medical center genexpert labora...   
382                 zamboanga del norte medical center   
383    zamboanga del sur medical center pcr laboratory   
384                         truhealthencoder@gmail.com   

     Cumulative Samples Tested  
0                        86372  
1                        27634  
2                       235188  
3                         7292  
4                          262  
..                         ...  
380                      98686  
381                      56578  
382                      15342  
383                      44188  
384                         12  

[385 rows x 2 columns]

In [120]:
i = 0
for facility in tests_df['Facility']:
    if facility not in facility_df['Facility'].values:
        print(i, facility)
        i += 1

0 truhealthencoder@gmail.com


Only records with facility name "truhealthencoder@gmail.com" will not be included.

In [122]:
merged_df = facility_df.merge(tests_df, on='Facility')

In [123]:
merged_df

Facility             Province  \
0                                  a star laboratories         METRO MANILA   
1                   ac molecular diagnostic laboratory         METRO MANILA   
2               adela serra ty memorial medical center      SURIGAO DEL SUR   
3                   al molecular diagnostic laboratory         METRO MANILA   
4       al molecular diagnostic laboratory - genexpert         METRO MANILA   
..                                                 ...                  ...   
381  zamboanga city medical center genexpert labora...    ZAMBOANGA DEL SUR   
382       zamboanga city medical center  (plate-based)    ZAMBOANGA DEL SUR   
383  zamboanga city medical center - da satellite l...    ZAMBOANGA DEL SUR   
384                 zamboanga del norte medical center  ZAMBOANGA DEL NORTE   
385    zamboanga del sur medical center pcr laboratory    ZAMBOANGA DEL SUR   

     Cumulative Samples Tested  
0                        86372  
1                        27634  
2                        84410  
3                       235188  
4                         7292  
..                         ...  
381                      56578  
382                       1765  
383                      98686  
384                      15342  
385                      44188  

[386 rows x 3 columns]

In [127]:
merged_df[merged_df['Facility'].isna()]

Empty DataFrame
Columns: [Facility, Province, Cumulative Samples Tested]
Index: []

In [128]:
merged_df[merged_df['Province'].isna()]

Empty DataFrame
Columns: [Facility, Province, Cumulative Samples Tested]
Index: []

In [129]:
merged_df[merged_df['Cumulative Samples Tested'].isna()]

Empty DataFrame
Columns: [Facility, Province, Cumulative Samples Tested]
Index: []

In [130]:
output_file = "/Users/jimuelcelestejr/Documents/codebook/covid-asmr-philippines/data/COVID-19 Tests/covid19_tests_performed_by_facility.csv"
merged_df.to_csv(output_file, index=False)

In [135]:
tests_province_df = merged_df.groupby('Province').sum().reset_index()

/var/folders/ny/jv8cwgpx7ng41dfgnqsvzdnw0000gn/T/ipykernel_23663/889095452.py:1: FutureWarning: The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  tests_province_df = merged_df.groupby('Province').sum().reset_index()


In [137]:
tests_province_df = tests_province_df.rename(columns={'Province': 'PROVINCE'})
tests_province_df

PROVINCE  Cumulative Samples Tested
0                   AGUSAN DEL SUR                     144854
1                            AKLAN                      76939
2                            ALBAY                     290687
3                          ANTIQUE                      25303
4                           BATAAN                     291508
5                         BATANGAS                     557066
6                          BENGUET                     936308
7                            BOHOL                     143963
8                         BUKIDNON                      10359
9                          BULACAN                     544194
10                         CAGAYAN                     461277
11                  CAGAYAN DE ORO                     217631
12                          CAINTA                      35033
13                         CALINGA                      19949
14                   CAMARINES SUR                      62175
15                        CAMIGUIN                        439
16                           CAPIZ                      46712
17                          CAVITE                     719926
18                           CEBU                     2096294
19                        COTABATO                     297446
20  COTABATO CITY (NOT A PROVINCE)                       7217
21                    DAVAO DE ORO                      44200
22                 DAVAO DEL NORTE                     167912
23                   DAVAO DEL SUR                    1352364
24                   ILOCOS NORTE                      250855
25                      ILOCOS SUR                       6132
26                          ILOILO                    1122768
27                         ISABELA                     155289
28                        LA UNION                     578394
29                          LAGUNA                     536618
30                 LANAO DEL NORTE                      27754
31                   LANAO DEL SUR                      83750
32                           LEYTE                     510102
33                     MAGUINDANAO                        923
34                      MARINDUQUE                       2305
35                    METRO MANILA                   18963382
36                         MINDORO                      14605
37              MISAMIS OCCIDENTAL                      26391
38                MISAMIS ORIENTAL                     331182
39               MOUNTAIN PROVINCE                      38469
40               NEGROS OCCIDENTAL                     740786
41                 NEGROS ORIENTAL                      53322
42                  NORTH COTABATO                       3950
43                     NUEVA ECIJA                      46905
44                ORIENTAL MINDORO                      29725
45                         PALAWAN                      53745
46                        PAMPANGA                    1199092
47                      PANGASINAN                     242627
48                          QUEZON                     227026
49                           RIZAL                      60117
50                       SORSOGON                       16590
51                  SOUTH COTABATO                     140956
52                  SOUTHERN LEYTE                       2529
53                  SULTAN KUDARAT                      28337
54               SURIGAO DEL NORTE                     181822
55                 SURIGAO DEL SUR                      84410
56                          TARLAC                      80388
57                        ZAMBALES                     231793
58             ZAMBOANGA DEL NORTE                      25898
59               ZAMBOANGA DEL SUR                     475455

In [138]:
output_file = "/Users/jimuelcelestejr/Documents/codebook/covid-asmr-philippines/data/COVID-19 Tests/covid19_tests_performed_by_province.csv"
tests_province_df.to_csv(output_file, index=False)

In [139]:
tests_province_df.shape

(60, 2)

Only 60 provinces have recorded testing facilities.

## COVID-19 Tests Performed per 100,000 population

In [143]:
pop = "/Users/jimuelcelestejr/Documents/codebook/covid-asmr-philippines/data/Philippine Population 2020 by Province and Age Group.xlsx"
pop = pd.read_excel(pop, index_col=0)

In [144]:
pop

0 to 4  5 to 9  10 to 14  15 to 19  20 to 24  \
BASILAN                          63979   61636     56582     45050     40105   
COTABATO CITY (NOT A PROVINCE)   38216   35974     34523     35957     36567   
LANAO DEL SUR                   182925  160576    156233    137554    123810   
MAGUINDANAO                     223251  193100    179050    138736    124325   
SULU                            130431  135390    130518    122121    100731   
...                                ...     ...       ...       ...       ...   
DAVAO ORIENTAL                   59339   64443     64640     58727     50134   
COTABATO (NORTH COTABATO)       133770  145175    141983    129362    111947   
SARANGANI                        64989   66801     64498     59084     51538   
SOUTH COTABATO                  176489  180879    175577    161973    157207   
SULTAN KUDARAT                   94981   98354     95760     91007     81129   

                                25 to 29  30 to 34  35 to 39  40 to 44  \
BASILAN                            34417     27226     23655     19508   
COTABATO CITY (NOT A PROVINCE)     32404     24889     20741     17727   
LANAO DEL SUR                     101418     75871     63633     55861   
MAGUINDANAO                       106176     89110     76435     66797   
SULU                               84344     62854     58477     50591   
...                                  ...       ...       ...       ...   
DAVAO ORIENTAL                     43254     39477     33234     31758   
COTABATO (NORTH COTABATO)         103607     93164     80130     72339   
SARANGANI                          46011     39592     33812     30148   
SOUTH COTABATO                    144622    129035    109715    100470   
SULTAN KUDARAT                     70898     62278     52509     46940   

                                45 to 49  50 to 54  55 to 59  60 to 64  \
BASILAN                            16292     12212      8616      6256   
COTABATO CITY (NOT A PROVINCE)     13937     10973      7678      5916   
LANAO DEL SUR                      44624     32651     20800     15346   
MAGUINDANAO                        50553     34625     22091     15534   
SULU                               42671     28650     18751     13046   
...                                  ...       ...       ...       ...   
DAVAO ORIENTAL                     29504     26855     22762     18565   
COTABATO (NORTH COTABATO)          63413     55740     44849     36238   
SARANGANI                          26700     21972     17369     13518   
SOUTH COTABATO                     85317     72439     57481     44890   
SULTAN KUDARAT                     40499     34207     26326     20979   

                                65 to 69  70 to 74  75 to 79    80+  
BASILAN                             4528      2411      1320   1318  
COTABATO CITY (NOT A PROVINCE)      3891      2535      1239   1372  
LANAO DEL SUR                      10967      5999      3125   3114  
MAGUINDANAO                        10130      5886      3036   2979  
SULU                                9290      5469      2727   2614  
...                                  ...       ...       ...    ...  
DAVAO ORIENTAL                     13140      8676      5220   5256  
COTABATO (NORTH COTABATO)          25874     16176      9899   9928  
SARANGANI                           9355      5791      3524   3367  
SOUTH COTABATO                     31244     19120     11349  10749  
SULTAN KUDARAT                     14809      9593      5537   5748  

[85 rows x 17 columns]

In [147]:
pop = pop.sum(axis=1)
pop

BASILAN                            425111
COTABATO CITY (NOT A PROVINCE)     324539
LANAO DEL SUR                     1194507
MAGUINDANAO                       1341814
SULU                               998675
                                   ...   
DAVAO ORIENTAL                     574984
COTABATO (NORTH COTABATO)         1273594
SARANGANI                          558069
SOUTH COTABATO                    1668556
SULTAN KUDARAT                     851554
Length: 85, dtype: int64

In [152]:
pop = pop.reset_index().rename(columns={'index': 'PROVINCE', 0: 'Population'})
pop

PROVINCE  Population
0                          BASILAN      425111
1   COTABATO CITY (NOT A PROVINCE)      324539
2                    LANAO DEL SUR     1194507
3                      MAGUINDANAO     1341814
4                             SULU      998675
..                             ...         ...
80                  DAVAO ORIENTAL      574984
81       COTABATO (NORTH COTABATO)     1273594
82                       SARANGANI      558069
83                  SOUTH COTABATO     1668556
84                  SULTAN KUDARAT      851554

[85 rows x 2 columns]

In [154]:
pop.merge(tests_province_df, on='PROVINCE')

PROVINCE  Population  Cumulative Samples Tested
0   COTABATO CITY (NOT A PROVINCE)      324539                       7217
1                    LANAO DEL SUR     1194507                      83750
2                      MAGUINDANAO     1341814                        923
3                          BENGUET      822619                     936308
4                MOUNTAIN PROVINCE      157798                      38469
5                   AGUSAN DEL SUR      737991                     144854
6                SURIGAO DEL NORTE      531753                     181822
7                  SURIGAO DEL SUR      640512                      84410
8                       ILOCOS SUR      704218                       6132
9                         LA UNION      820343                     578394
10                      PANGASINAN     3159228                     242627
11                         CAGAYAN     1265540                     461277
12                         ISABELA     1695539                     155289
13                          BATAAN      849575                     291508
14                         BULACAN     3696937                     544194
15                     NUEVA ECIJA     2306751                      46905
16                        PAMPANGA     2894309                    1199092
17                          TARLAC     1499064                      80388
18                        ZAMBALES      906184                     231793
19                        BATANGAS     2902855                     557066
20                          CAVITE     4318663                     719926
21                          LAGUNA     3373136                     536618
22                          QUEZON     2223791                     227026
23                           RIZAL     3321325                      60117
24                      MARINDUQUE      238830                       2305
25                ORIENTAL MINDORO      906661                      29725
26                         PALAWAN     1237280                      53745
27             ZAMBOANGA DEL NORTE     1046017                      25898
28               ZAMBOANGA DEL SUR     2017793                     475455
29                           ALBAY     1372550                     290687
30                   CAMARINES SUR     2062277                      62175
31                           AKLAN      612985                      76939
32                         ANTIQUE      611478                      25303
33                           CAPIZ      803879                      46712
34                          ILOILO     2503326                    1122768
35               NEGROS OCCIDENTAL     3216287                     740786
36                           BOHOL     1390524                     143963
37                 NEGROS ORIENTAL     1428548                      53322
38                           LEYTE     2020426                     510102
39                  SOUTHERN LEYTE      428484                       2529
40                        BUKIDNON     1537629                      10359
41                        CAMIGUIN       92696                        439
42                 LANAO DEL NORTE     1083898                      27754
43              MISAMIS OCCIDENTAL      614951                      26391
44                MISAMIS ORIENTAL     1678624                     331182
45                    DAVAO DE ORO      766299                      44200
46                 DAVAO DEL NORTE     1115167                     167912
47                   DAVAO DEL SUR     2450445                    1352364
48                  SOUTH COTABATO     1668556                     140956
49                  SULTAN KUDARAT      851554                      28337

In [156]:
i = 0
for province in pop['PROVINCE']:
    if province not in tests_province_df['PROVINCE'].values:
        print(i, province)
        i += 1

0 BASILAN
1 SULU
2 TAWI-TAWI
3 INTERIM PROVINCE
4 ABRA
5 APAYAO
6 IFUGAO
7 KALINGA
8 AGUSAN DEL NORTE
9 DINAGAT ISLANDS
10 NCR
11 ILOCOS NORTE
12 BATANES
13 NUEVA VIZCAYA
14 QUIRINO
15 AURORA
16 OCCIDENTAL MINDORO
17 ROMBLON
18 CITY OF ISABELA (NOT A PROVINCE)
19 ZAMBOANGA SIBUGAY
20 CAMARINES NORTE
21 CATANDUANES
22 MASBATE
23 SORSOGON
24 GUIMARAS
25 CEBU
26 SIQUIJOR
27 BILIRAN
28 EASTERN SAMAR
29 NORTHERN SAMAR
30 SAMAR (WESTERN SAMAR)
31 DAVAO OCCIDENTAL
32 DAVAO ORIENTAL
33 COTABATO (NORTH COTABATO)
34 SARANGANI


35 provinces do not have testing facility information. Some manual editing is necessary.

1. Check spellings / additional information for each province.

In [158]:
i = 0
for province in tests_province_df['PROVINCE']:
    if province not in pop['PROVINCE'].values:
        print(i, province)
        i += 1

0 CAGAYAN DE ORO
1 CAINTA
2 CALINGA
3 CEBU 
4 COTABATO
5 ILOCOS NORTE 
6 METRO MANILA
7 MINDORO
8 NORTH COTABATO
9 SORSOGON 


10 provinces are in test that are not in the population. It might be a spelling matter. Need some editing.

In [163]:
renaming_map = {
    'CAGAYAN DE ORO': 'MISAMIS ORIENTAL',
    'CAINTA': 'RIZAL',
    'CALINGA': 'KALINGA', 
    'CEBU ': 'CEBU', 
    'COTABATO': 'COTABATO CITY (NOT A PROVINCE)',
    'ILOCOS NORTE ': 'ILOCOS NORTE', 
    'METRO MANILA': 'NCR', 
    'MINDORO': 'ORIENTAL MINDORO',
    'NORTH COTABATO': 'COTABATO (NORTH COTABATO)',
    'SORSOGON ': 'SORSOGON'
}

I am manually editing file ```covid19_tests_performed_by_province.csv``` to reflect the changes as defined here.

Example: Adding Cagayan de Oro to Misamis Oriental and deleting Cagayan de Oro from the list.
See: covid19_tests_performed_by_province_manually_modified.csv

In [169]:
tests_province_df_2 = "/Users/jimuelcelestejr/Documents/codebook/covid-asmr-philippines/data/COVID-19 Tests/covid19_tests_performed_by_province_manually_modified.csv"
tests_province_df_2 = pd.read_csv(tests_province_df_2)

In [170]:
tests_province_df_2

PROVINCE  Cumulative Samples Tested
0                   AGUSAN DEL SUR                     144854
1                            AKLAN                      76939
2                            ALBAY                     290687
3                          ANTIQUE                      25303
4                           BATAAN                     291508
5                         BATANGAS                     557066
6                          BENGUET                     936308
7                            BOHOL                     143963
8                         BUKIDNON                      10359
9                          BULACAN                     544194
10                         CAGAYAN                     461277
11                         KALINGA                      19949
12                   CAMARINES SUR                      62175
13                        CAMIGUIN                        439
14                           CAPIZ                      46712
15                          CAVITE                     719926
16                            CEBU                    2096294
17  COTABATO CITY (NOT A PROVINCE)                     304663
18                    DAVAO DE ORO                      44200
19                 DAVAO DEL NORTE                     167912
20                   DAVAO DEL SUR                    1352364
21                    ILOCOS NORTE                     250855
22                      ILOCOS SUR                       6132
23                          ILOILO                    1122768
24                         ISABELA                     155289
25                        LA UNION                     578394
26                          LAGUNA                     536618
27                 LANAO DEL NORTE                      27754
28                   LANAO DEL SUR                      83750
29                           LEYTE                     510102
30                     MAGUINDANAO                        923
31                      MARINDUQUE                       2305
32                             NCR                   18963382
33              MISAMIS OCCIDENTAL                      26391
34                MISAMIS ORIENTAL                     548813
35               MOUNTAIN PROVINCE                      38469
36               NEGROS OCCIDENTAL                     740786
37                 NEGROS ORIENTAL                      53322
38       COTABATO (NORTH COTABATO)                       3950
39                     NUEVA ECIJA                      46905
40                ORIENTAL MINDORO                      44330
41                         PALAWAN                      53745
42                        PAMPANGA                    1199092
43                      PANGASINAN                     242627
44                          QUEZON                     227026
45                           RIZAL                      95150
46                        SORSOGON                      16590
47                  SOUTH COTABATO                     140956
48                  SOUTHERN LEYTE                       2529
49                  SULTAN KUDARAT                      28337
50               SURIGAO DEL NORTE                     181822
51                 SURIGAO DEL SUR                      84410
52                          TARLAC                      80388
53                        ZAMBALES                     231793
54             ZAMBOANGA DEL NORTE                      25898
55               ZAMBOANGA DEL SUR                     475455

In [171]:
i = 0
for province in tests_province_df_2['PROVINCE']:
    if province not in pop['PROVINCE'].values:
        print(i, province)
        i += 1

All provinces in test are renamed accordingly, i.e., reflecting population df.

In [172]:
pop

PROVINCE  Population
0                          BASILAN      425111
1   COTABATO CITY (NOT A PROVINCE)      324539
2                    LANAO DEL SUR     1194507
3                      MAGUINDANAO     1341814
4                             SULU      998675
..                             ...         ...
80                  DAVAO ORIENTAL      574984
81       COTABATO (NORTH COTABATO)     1273594
82                       SARANGANI      558069
83                  SOUTH COTABATO     1668556
84                  SULTAN KUDARAT      851554

[85 rows x 2 columns]

In [175]:
province_pop_test_df = pop.merge(tests_province_df_2, on='PROVINCE', how='left')
province_pop_test_df

PROVINCE  Population  Cumulative Samples Tested
0                          BASILAN      425111                        NaN
1   COTABATO CITY (NOT A PROVINCE)      324539                   304663.0
2                    LANAO DEL SUR     1194507                    83750.0
3                      MAGUINDANAO     1341814                      923.0
4                             SULU      998675                        NaN
..                             ...         ...                        ...
80                  DAVAO ORIENTAL      574984                        NaN
81       COTABATO (NORTH COTABATO)     1273594                     3950.0
82                       SARANGANI      558069                        NaN
83                  SOUTH COTABATO     1668556                   140956.0
84                  SULTAN KUDARAT      851554                    28337.0

[85 rows x 3 columns]

In [178]:
province_pop_test_df['Cumulative Samples Tested'] = province_pop_test_df['Cumulative Samples Tested'].fillna(0)
province_pop_test_df

PROVINCE  Population  Cumulative Samples Tested
0                          BASILAN      425111                        0.0
1   COTABATO CITY (NOT A PROVINCE)      324539                   304663.0
2                    LANAO DEL SUR     1194507                    83750.0
3                      MAGUINDANAO     1341814                      923.0
4                             SULU      998675                        0.0
..                             ...         ...                        ...
80                  DAVAO ORIENTAL      574984                        0.0
81       COTABATO (NORTH COTABATO)     1273594                     3950.0
82                       SARANGANI      558069                        0.0
83                  SOUTH COTABATO     1668556                   140956.0
84                  SULTAN KUDARAT      851554                    28337.0

[85 rows x 3 columns]

In [179]:
province_pop_test_df['Tests / Population'] = province_pop_test_df['Cumulative Samples Tested'] / province_pop_test_df['Population']
province_pop_test_df

PROVINCE  Population  Cumulative Samples Tested  \
0                          BASILAN      425111                        0.0   
1   COTABATO CITY (NOT A PROVINCE)      324539                   304663.0   
2                    LANAO DEL SUR     1194507                    83750.0   
3                      MAGUINDANAO     1341814                      923.0   
4                             SULU      998675                        0.0   
..                             ...         ...                        ...   
80                  DAVAO ORIENTAL      574984                        0.0   
81       COTABATO (NORTH COTABATO)     1273594                     3950.0   
82                       SARANGANI      558069                        0.0   
83                  SOUTH COTABATO     1668556                   140956.0   
84                  SULTAN KUDARAT      851554                    28337.0   

    Tests / Population  
0             0.000000  
1             0.938756  
2             0.070113  
3             0.000688  
4             0.000000  
..                 ...  
80            0.000000  
81            0.003101  
82            0.000000  
83            0.084478  
84            0.033277  

[85 rows x 4 columns]

In [180]:
province_pop_test_df['Tests / Population per 100k (2019-2023)'] = province_pop_test_df['Tests / Population'] * 100000
province_pop_test_df

PROVINCE  Population  Cumulative Samples Tested  \
0                          BASILAN      425111                        0.0   
1   COTABATO CITY (NOT A PROVINCE)      324539                   304663.0   
2                    LANAO DEL SUR     1194507                    83750.0   
3                      MAGUINDANAO     1341814                      923.0   
4                             SULU      998675                        0.0   
..                             ...         ...                        ...   
80                  DAVAO ORIENTAL      574984                        0.0   
81       COTABATO (NORTH COTABATO)     1273594                     3950.0   
82                       SARANGANI      558069                        0.0   
83                  SOUTH COTABATO     1668556                   140956.0   
84                  SULTAN KUDARAT      851554                    28337.0   

    Tests / Population  Tests / Population per 100k (2019-2023)  
0             0.000000                                 0.000000  
1             0.938756                             93875.620496  
2             0.070113                              7011.260713  
3             0.000688                                68.787477  
4             0.000000                                 0.000000  
..                 ...                                      ...  
80            0.000000                                 0.000000  
81            0.003101                               310.145933  
82            0.000000                                 0.000000  
83            0.084478                              8447.783593  
84            0.033277                              3327.680922  

[85 rows x 5 columns]

In [182]:
output_file = "/Users/jimuelcelestejr/Documents/codebook/covid-asmr-philippines/data/COVID-19 Tests/tests_by_province_per_100k.csv"
province_pop_test_df.to_csv(output_file, index=False)